docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,326 @@
+
+VPA is used to scale `pvc-autoscaler`. For the future security enhancement scenario, where `kube-rbac-proxy` is used
+(see [section 'Future Enhancements'](#future-enhancements)),
+that container is expected to be excluded from autoscaling, and to run with fixed resource requests. Pod evictions
+driven by that lean container are unjustified, as they would only achieve minimal resource savings, and disrupt
+the main container.
+
+The feature is deployed behind a dedicated feature gate. When disabled, `pvc-autoscaler` is removed, and related
+annotations are removed from the shoot control plane observability StatefulSets, and the seed observability StatefulSets.
+
+#### Initial volume size and MaxAllowed
+
+| Component type | Component            | Old size | Initial size    | Max size |
+|----------------|----------------------|----------|-----------------|----------|
+| Shoot          | prometheus           | 20Gi     | 1Gi             | 40Gi     |
+| Shoot          | vali                 | 30Gi     | 1Gi             | 60Gi     |
+| Seed           | prometheus-aggregate | 20Gi     | 2Gi             | 40Gi     |
+| Seed           | prometheus-cache     | 10Gi     | 6Gi<sup>*</sup> | 20Gi     |
+| Seed           | prometheus-seed      | 100Gi    | 5Gi             | 200Gi    |
+| Seed           | vali                 | 100Gi    | 5Gi             | 200Gi    |
+
+<sup>*</sup> The greater initial size mitigates [risk due to circular dependency](#reliance-on-prometheus). 
+### Transition Strategy
+The following steps will be executed over time, to minimise the risk of disruption to existing Gardener instances:
+
+1. The proposed scaling approach will be deployed behind a feature gate and disabled by default.
+2. The new feature will be gradually enabled on seed-by-seed basis:
+   1. The feature gate will be enabled for the seed
+   2. The correct operation of newly created shoots will be verified via direct observation
+   3. The observability PVCs of preexisting shoots will gradually be annotated for scaling by `pvc-autoscaler`
+   4. The correct operation of preexisting shoots will be verified via direct observation
+3. The feature flag will be promoted and eventually removed.
+
+The following metrics will be recorded before and/or after the transition of the seed to the new functionality,
+in order to evaluate the economical effect, and the reliability of the scaling solution:
+   - Average utilisation of observability volumes:
+     - Preexisting volumes (before and after)
+     - Newly created volumes (after)
+   - Percentage of observability volumes above 90% utilisation (before and after)
+   - Utilisation of observability volumes at p10, p50, p90:
+     - Preexisting volumes (before and after)
+     - Newly created volumes (after)
+   - Peak volume utilisation (after)
+   - Observability volume size: at p10, p50, p90, and average size (after)
+   - Observability volume average size and count (before)
+
+### Future Enhancements
```

Comment:
What about plans to scale ETCD and shoot volumes? The headline says "Phase 1", so there are other phases and more plans?

Reply:
TL;DR: Next phase = PVC autoscaling capability for customer (shoot) volumes.

Providing autoscaling capability to shoot PVs absolutely is in the vision. We don't have a fixed execution plan yet, but we're quite certain that we want it. The high level "plan" is, deploy for internal observability volumes (current phase), use that to gain confidence/maturity, then offer the functionality to customer (shoot) PVs.

I haven't considered ETCD, so I don't know yet if it would be a good candidate. I know that it has an 8GB limit to the data it can store, but I guess the WAL file could grow substantially larger, under heavy load and delayed compactions. On the other hand, it may be wasteful to inflate the volume due to a one-time high-load event inflating the WAL, and then pay for a large volume ever after.

Reply:
@andrerun: Please add these thoughts to the document :)
@unmarshall @shreyas-s-rao and colleagues: Thoughts about scaling the ETCD volume?

Reply:
You can have a look at:
https://github.com/gardener/etcd-druid/issues/481#issuecomment-1346402013

Reply:
Done. Notes regarding future plans (scaling customer volumes, and not scaling Etcd volumes), are now available in the paragraph starting with "The current initiative is limited to scaling observability components", and the paragraph right after it.